File: frontend/types/document.ts
// Mirrors backend/app/schemas/document.py — Phase 9.2 real pipeline fields.

export type DocumentType = "contract" | "evidence" | "correspondence" | "generated" | "other";

export type DocumentStatus = "uploaded" | "processing" | "ready" | "failed" | "ocr_required" | "unsupported";

export interface Document {
  id: string;
  workspace_id: string;
  title: string;
  document_type: DocumentType;
  original_filename: string | null;
  media_type: string | null;
  size_bytes: number | null;
  sha256: string | null;
  status: DocumentStatus;
  processing_error: string | null;
  created_at: string | null;
  processed_at: string | null;
  doc_metadata: {
    extractor?: string;
    page_count?: number;
    chunk_count?: number;
    used_structure_detection?: boolean;
    warnings?: string[];
  };
}

export interface DocumentCitation {
  citation_type: string;
  document_id: string;
  document_title: string;
  page_number: number | null;
  section_path: string | null;
  excerpt: string;
  label: string;
  chunk_id: string | null;
  content_hash: string | null;
}

export interface DocumentAskResponse {
  status: "answered" | "insufficient_document_evidence";
  answer: string;
  citations: DocumentCitation[];
  // "extractive" = deterministic regex match, no LLM call; "llm" = evidence-gated LLM reasoning.
  answer_method: "extractive" | "llm";
}

export interface ExtractedFact {
  value: string;
  provenance: string;
  kind: "date" | "amount" | "party";
}

export interface DocumentAnalyzeResponse {
  status: "analyzed" | "insufficient_document_evidence";
  document_type_extracted: string | null;
  extracted_dates: ExtractedFact[];
  extracted_amounts: ExtractedFact[];
  extracted_parties: ExtractedFact[];
  inferred_obligations: string[];
  inferred_risks: string[];
  inferred_missing_information: string[];
}

File: backend/app/schemas/document.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.matters import DocumentStatus, DocumentType


class DocumentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    workspace_id: uuid.UUID
    title: str
    document_type: DocumentType
    original_filename: str | None
    media_type: str | None
    size_bytes: int | None
    sha256: str | None
    status: DocumentStatus
    processing_error: str | None
    created_at: datetime | None = None
    processed_at: datetime | None = None
    doc_metadata: dict = {}


class DocumentAskRequest(BaseModel):
    question: str


class DocumentCitationOut(BaseModel):
    citation_type: str
    document_id: uuid.UUID
    document_title: str
    page_number: int | None
    section_path: str | None
    excerpt: str
    label: str
    chunk_id: uuid.UUID | None = None
    content_hash: str | None = None


class DocumentAskResponse(BaseModel):
    status: str
    answer: str
    citations: list[DocumentCitationOut]
    answer_method: str = "llm"


class ExtractedFactOut(BaseModel):
    value: str
    provenance: str
    kind: str


class DocumentAnalyzeResponse(BaseModel):
    status: str
    document_type_extracted: str | None
    extracted_dates: list[ExtractedFactOut]
    extracted_amounts: list[ExtractedFactOut]
    extracted_parties: list[ExtractedFactOut]
    inferred_obligations: list[str]
    inferred_risks: list[str]
    inferred_missing_information: list[str]
